plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteJdbcClient.java

```diff
@@ -0,0 +1,562 @@
+                        if (backups > 1) {
+                            properties.put(property, backups);
+                        }
+                        break;
+                    case IgniteTableProperties.TEMPLATE_PROPERTY:
+                        Optional.ofNullable(resultSet.getString(property.toUpperCase(ENGLISH)))
+                                .map(IgniteTemplateType::valueOf)
+                                .filter(template -> template != IgniteTemplateType.PARTITIONED)
+                                .ifPresent(value -> properties.put(property, value));
+                        break;
+                    case IgniteTableProperties.WRITE_SYNCHRONIZATION_MODE_PROPERTY:
+                        Optional.ofNullable(resultSet.getString(property.toUpperCase(ENGLISH)))
+                                .map(IgniteWriteSyncMode::valueOf)
+                                .filter(mode -> mode != IgniteWriteSyncMode.FULL_SYNC)
+                                .ifPresent(value -> properties.put(property, value));
+                        break;
+                    case IgniteTableProperties.CACHE_NAME_PROPERTY:
+                    case IgniteTableProperties.CACHE_GROUP_PROPERTY:
+                        Optional.ofNullable(resultSet.getString(property.toUpperCase(ENGLISH)))
+                                .filter(name -> !name.equals("SQL_PUBLIC_" + tableName))
+                                .ifPresent(value -> properties.put(property, value));
+                        break;
+                    case IgniteTableProperties.DATA_REGION_PROPERTY:
+                        Optional.ofNullable(resultSet.getString(property.toUpperCase(ENGLISH))).ifPresent(value -> properties.put(property, value));
+                        break;
+                    default:
+                        throw new IllegalStateException("Unexpected value: " + property);
+                }
+            }
+        }
+        catch (SQLException e) {
+            throw new TrinoException(JDBC_ERROR, e);
+        }
+        return properties.build();
+    }
+
+    // extract result from format: "ID" ASC, "CITY_ID" ASC
+    private List<String> extract(String row)
+    {
+        ImmutableList.Builder<String> builder = ImmutableList.builder();
+        if (isNullOrEmpty(row)) {
+            return builder.build();
+        }
+        for (String key : row.split(",")) {
+            int left = key.indexOf("\"") + 1;
+            int right = key.lastIndexOf("\"");
+            builder.add(key.substring(left, right).toLowerCase(ENGLISH));
+        }
+        return builder.build();
+    }
+
+    @Override
+    public WriteMapping toWriteMapping(ConnectorSession session, Type type)
+    {
+        if (type instanceof DateType) {
+            return WriteMapping.longMapping("date", dateWriteFunction());
+        }
+        if (type instanceof DecimalType) {
```

Comment:
please move above date case

genmerally the preferred order of handling types is

- boolean
- tinyint, smallint, integer, bigint
- real, double
- decimal
- char, varchar
- varbinary
- date
- time, time with time zone
- timestamp, timestamp with time zone
- ...